include/osg/BufferIndexBinding

```diff
@@ -213,6 +214,75 @@ class OSG_EXPORT ShaderStorageBufferBinding : public BufferIndexBinding
     }
 };
 
+
+///Read back BufferData from Device
+class SyncBufferDataCallback;
+class BufferDataReadBack : public Camera::DrawCallback{
+public:
+    BufferDataReadBack(SyncBufferDataCallback* u=0): Camera::DrawCallback(), _upcb(u){}
+    inline void setSyncBufferDataUpdateCallback( SyncBufferDataCallback *u){ _upcb=u; }
+
+    virtual void operator () (osg::RenderInfo& renderInfo) const;
+
+    /** read back BufferObject to its BufferData (reimplement for custom behavior) */
+    virtual void readBackBufferData(osg::RenderInfo& renderInfo) const;
+protected:
+    SyncBufferDataCallback* _upcb;
```

Comment:
This is just asking for a dangling pointer issue.  From a first run through I haven't been able to see the whole thing clearly enough to suggest exactly what you need to resolve it.  It's not safe as is so will need some form of resolution before it's safe to merge.

Reply:
the _upcb is a weak pointer as it's setted during SyncBufferDataCallback update traversal.
in my last commit, i removed public setter and made class BufferDataReadBack friend of SyncBufferDataCallback in order it access protected _upcb